compiler/vm.nim

```diff
@@ -2205,6 +2205,13 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
           stackTrace(c, tos, pc, "key does not exist: " & key)
       else:
         stackTrace(c, tos, pc, "key does not exist: " & destKey)
+    of opcNctContains:
```

Comment:
Can this be implemented as a callback instead?

Reply:
I think it is feasible since `PCtx` is accessible in function `proc registerAdditionalOps*(c: PCtx)`.

Reply:
Not too sure if I implemented it correctly but got it working as a callback

Using this quickly whipped up benchmark
```nim
import std/[times, macros, macrocache, math]

const
  t = CacheTable"l"
  n = 1_000

template timeIt(title: string, body: untyped) =
  var times: array[10, float]
  for i in 0..<10:
    let start = cpuTime()
    for i in 0..<n:
      body
    times[i] = cpuTime() - start
  echo title, ": ", sum(times) / len(times).float

proc main() =
  timeIt "Using magic":
    doAssert t.hasKeyMag("test")

  timeIt "Using VM hook":
    doAssert t.hasKey("test")

static:
  t["test"] = newLit 8
  main()
```
I get these numbers
```
Using magic: 0.0008407110000000717
Using VM hook: 0.001223563300000041
```

Reply:
Nice!